c++/src/kj/compat/tls.c++

```diff
@@ -162,11 +163,20 @@ public:
       }
     });
   }
+
   kj::Promise<void> accept() {
     // We are the server. Set SSL options to prefer server's cipher choice.
     SSL_set_options(ssl, SSL_OP_CIPHER_SERVER_PREFERENCE);
 
-    return sslCall([this]() { return SSL_accept(ssl); }).ignoreResult();
+    auto acceptPromise = sslCall([this]() {
+      return SSL_accept(ssl);
+    });
+    return acceptPromise.then([](size_t ret) {
+      if (ret == 0) {
+        kj::throwRecoverableException(
+            KJ_EXCEPTION(DISCONNECTED, "Client disconnected during SSL_accept()"));
```

Comment:
This will eventually get `KJ_LOG()`ed in `TlsConnectionReceiver::taskFailed()`, but this doesn't seem like an error that we want generating Sentry reports. Maybe `taskFailed()` should avoid logging `DISCONNECTED` exceptions?

Reply:
I'm concerned that we may lack any other avenue to detect a burst of aborted TLS connection attempts. Would you be okay if I keep as is but drop a TODO(someday) explaining that we may want to do this in the future?

Reply:
In general I would say that aborted connection attempts are really up to the client to report -- the server does not know if it's an error or not, so shouldn't report it as one. We could arguably add a metric to count these, so that we could alert on sudden spikes. But reporting a Sentry error is definitely not what we want here. We know for a fact that we have security scanners and such which are making these connections and dropping them today, and that's not a problem, but our goal with Sentry is to avoid reporting anything that isn't actionable (which, obviously, we're quite a ways from today, but that's the goal). So we really can't report these to Sentry.

Reply:
Hmmmm, in my experience, I have seen logs on both sides prove to be valuable. I feel strongly that we shouldn't ignore these on the server side.

How about I add either a `kj::Maybe<kj::Own<TaskSet::ErrorHandler>>` or a `kj::Maybe<kj::Function<void(Exception&&)>>` to TLS context? Then the user can handle it how they want downstream.

In our case, I'd probably log in a why that does not report to Sentry.

Reply:
I'm OK with adding a new callback to handle connections that drop during the handshake, but can we plan on doing that in a separate PR? For now I think the best default behavior is to log only non-disconnected exceptions.